packages/flutter_tools/lib/src/migrations/swift_package_manager_integration_migration.dart

```diff
@@ -965,21 +947,13 @@ class ParsedProjectGroup {
 /// Representation of data parsed from PBXFrameworksBuildPhase section in Xcode
 /// project's project.pbxproj.
 class ParsedProjectFrameworksBuildPhase {
-  ParsedProjectFrameworksBuildPhase._(this.identifier, this.files);
-
-  factory ParsedProjectFrameworksBuildPhase.fromJson(
-      String key, Map<String, Object?> data) {
-    final List<String> parsedFiles = <String>[];
-    if (data['files'] is List<Object?>) {
-      for (final Object? item in data['files']! as List<Object?>) {
-        if (item is String) {
-          parsedFiles.add(item);
-        }
-      }
-      return ParsedProjectFrameworksBuildPhase._(key, parsedFiles);
-    }
-    return ParsedProjectFrameworksBuildPhase._(key, null);
-  }
+  ParsedProjectFrameworksBuildPhase.fromJson(
+    this.identifier,
+    Map<String, Object?> data,
+  ) : files = switch (data['files']) {
+        final List<Object?> files => files.whereType<String>().toList(),
+        _ => null,
+      };
```

Comment:
I do think this code is better than the original, since it saves us a redundant casting of `data['files']` to `List<Object?>`.